parser/gotest/internal/collector/collector.go

```diff
@@ -4,13 +4,24 @@ package collector
 
 import (
 	"sort"
+	"strings"
 	"time"
+	"unicode"
 )
 
 // line is a single line of output captured at some point in time.
 type line struct {
 	Timestamp time.Time
-	Text      string
+	text      string
+}
+
+func (l line) SafeText() string {
```

Comment:
The problem is with the XML output, so the parser is not really the right place to solve this. Let's modify the `formatOutput` function in the `junit` package instead and drop the control characters there.